node/engine/messageservice/p2p-message-service/dht-record.go

```diff
@@ -0,0 +1,85 @@
+package p2pms
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+const (
+	DHT_RECORD_PREFIX = "/scaddr/"
+)
+
+type stateChannelAddrToPeerIDValidator struct{}
+
+// DhtRecord represents the data stored in the DHT record
+type DhtRecord struct {
```

Comment:
Nit but since `DhtRecord` is only using in the `p2pms` package I think it could be a private struct. Same thing with `DhtData`
```suggestion
type dhtRecord struct {
```